regparser/layer/formatting.py

```diff
@@ -40,8 +40,14 @@ def build_header(xml_nodes):
     includes its colspan/rowspan"""
     stack = HeaderStack()
     stack.add(0, TableHeaderNode(None, 0))  # Root
+    ttitle_bump = 0
+    if len([_ for _ in xml_nodes if _.tag == "TTITLE"]):
+        ttitle_bump = 1
     for xml_node in xml_nodes:
-        level = int(xml_node.attrib['H'])
+        if xml_node.tag == "TTITLE":
+            level = 1
+        else:
+            level = int(xml_node.attrib['H']) + ttitle_bump
```

Comment:
What do you think about pulling this exceptional logic out of the loop? Maybe pass the `TTITLE` as another parameter to `build_header` or separate it out from the other `xml_nodes` before this loop? Then this would flow something like:
```python
if there_is_a_ttitle:
    add_element(ttitle)
    ttitle_bump = 1

for xml_node in other_xml_nodes:
    level = int(xml_node.attrib['H']) + ttitle_bump  # maybe even just int(there_is_a_ttitle)
    add_element(xml_node)
```
If we're going down the `CAPTION` route, we'll need this anyway, hey?